### PR TITLE
[aws] Add tags.yml file and upgrade to package spec 3.0

### PR DESCRIPTION
**Pull request by marc-gr**
## What does this PR do?

Add tags.yml file so that integration's dashboards and saved searches are tagged with "Security Solution" and displayed in the Security Solution UI.

As part of migrating to package spec 3.0.0 it removed duplicated, invalid field definitions, and invalid release properties.

Add missing dashboard filters.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates https://github.com/elastic/integrations/issues/7483
- Relates https://github.com/elastic/integrations/issues/7810

## Screenshots

image

### CODE REVIEW

**Review by kcreddy — APPROVED** [review]
LGTM
Just curios how `missing dashboard filters` is fixed? Did you run a script or had to apply the filters manually?

**Comment by kaiyan-sheng** [comment]
@tommyers-elastic I assume this PR will replace https://github.com/elastic/integrations/pull/7958 then?

File: packages/aws/manifest.yml [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
+version: 2.5.0
 description: Collect logs and metrics from Amazon Web Services (AWS) with Elastic Agent.
 type: integration
 categories:
   - aws
-release: ga
 conditions:
-  kibana.version: "^8.9.0"
+  kibana:
+    version: "^8.9.0"
```
Comment (kaiyan-sheng) [inline packages/aws/manifest.yml:11]: Based on https://github.com/elastic/package-spec/pull/567 kibana.version is equal or greater to 8.10.0?

File: packages/aws/manifest.yml [hunk 2]
```diff
@@ -1,15 +1,14 @@
-format_version: 1.0.0
+format_version: 3.0.0
 name: aws
 title: AWS
-version: 2.4.1
-license: basic
```
Comment (tommyers-elastic) [inline packages/aws/manifest.yml:5]: this should be copied to `conditions.elastic.subscription` as per https://github.com/elastic/elastic-package/blob/main/docs/howto/update_major_package_spec.md#field-root-additional-property-license-is-not-allowed

File: packages/aws/data_stream/s3access/_dev/test/pipeline/test-s3-server-access.log-expected.json [hunk 3]
```diff
@@ -25,6 +25,18 @@
             },
             "client": {
                 "address": "89.160.20.156",
+                "geo": {
```
Comment (tommyers-elastic) [inline packages/aws/data_stream/s3access/_dev/test/pipeline/test-s3-server-access.log-expected.json:28]: what's the reason these changes have ended up in the diff? it looks like all the fields are the same? or are there some subtle changes?

Comment (tommyers-elastic) [reply packages/aws/data_stream/s3access/_dev/test/pipeline/test-s3-server-access.log-expected.json:28]: oh wait, have they moved from root level into `client`?

File: packages/aws/data_stream/s3access/elasticsearch/ingest_pipeline/default.yml [hunk 4]
```diff
@@ -103,7 +103,7 @@ processors:
       ignore_empty_value: true
   - geoip:
       field: aws.s3access.remote_ip
-      target_field: geo
+      target_field: client.geo
```
Comment (tommyers-elastic) [inline packages/aws/data_stream/s3access/elasticsearch/ingest_pipeline/default.yml:106]: what's the reason for this change? isn't this a breaking change?

File: packages/aws/data_stream/s3access/_dev/test/pipeline/test-s3-server-access.log-expected.json (hunk 3 above)
Comment (tommyers-elastic) [reply packages/aws/data_stream/s3access/_dev/test/pipeline/test-s3-server-access.log-expected.json:28]: see comment below

File: packages/aws/kibana/dashboard/aws-3af47420-3e7b-11ea-bb0a-69c3ca1d410f.json [hunk 5]
```diff
@@ -3,7 +3,29 @@
         "description": "Logs AWS ELB Access Log Overview Dashboard",
         "kibanaSavedObjectMeta": {
             "searchSourceJSON": {
-                "filter": [],
+                "filter": [
```
Comment (tommyers-elastic) [inline packages/aws/kibana/dashboard/aws-3af47420-3e7b-11ea-bb0a-69c3ca1d410f.json:6]: with this addition, we can remove the filters which are currently set on each individual panel e.g. 

```                       
"mapStateJSON": "{\"center\":{\"lat\":50.97903,\"lon\":13.666},\"filters\":[{\"$state\":{\"store\":\"appState\"},\"meta\":{\"alias\":null,\"disabled\":false,\"index\":\"logs-*\",\"key\":\"data_stream.dataset\",\"negate\":false,\"params\":{\"query\":\"aws.elb_logs\"},\"type\":\"phrase\",\"value\":\"elb\"},\"query\":{\"match\":{\"data_stream.dataset\":{\"query\":\"aws.elb_logs\",\"type\":\"phrase\"}}}}],\"query\":{\"language\":\"kuery\",\"query\":\"\"},\"refreshConfig\":{\"interval\":0,\"isPaused\":false},\"timeFilters\":{\"from\":\"now-15m\",\"to\":\"now\"},\"zoom\":3.9,\"settings\":{\"autoFitToDataBounds\":false}}",
"title": "ELB Requests Geolocation [Logs AWS]",
```

File: packages/aws/kibana/dashboard/aws-fac28650-7349-11e9-816b-07687310a99a.json [hunk 6]
```diff
@@ -9,7 +9,168 @@
...
+                                "aws.s3_storage_lens",
+                                "aws.sns",
+                                "aws.sqs",
+                                "aws.transitgateway",
+                                "aws.usage",
+                                "aws.vpn"
+                            ],
+                            "type": "phrases"
+                        },
+                        "query": {
```
Comment (tommyers-elastic) [inline packages/aws/kibana/dashboard/aws-fac28650-7349-11e9-816b-07687310a99a.json:50]: would it make sense to just query for `aws.*`?

File: packages/aws/data_stream/s3access/elasticsearch/ingest_pipeline/default.yml (hunk 4 above)
Comment (marc-gr) [reply packages/aws/data_stream/s3access/elasticsearch/ingest_pipeline/default.yml:106]: It is breaking but this is fixing an error in the mapping, since `geo` fields are not allowed in the root as per ECS.

File: packages/aws/manifest.yml (hunk 1 above)
Comment (marc-gr) [reply packages/aws/manifest.yml:11]: We changed the constraint to avoid having to put minimum version 8.10 on all packages that want to add the tags. It will be just ignored on lower versions and a warning is shown at build/check time.

File: packages/aws/kibana/dashboard/aws-fac28650-7349-11e9-816b-07687310a99a.json (hunk 6 above)
Comment (marc-gr) [reply packages/aws/kibana/dashboard/aws-fac28650-7349-11e9-816b-07687310a99a.json:50]: I can change it, but this dashboard is supposed to only target metrics data streams, so it does not include any `type: logs` ones atm

File: packages/aws/data_stream/s3access/elasticsearch/ingest_pipeline/default.yml (hunk 4 above)
Comment (tommyers-elastic) [reply packages/aws/data_stream/s3access/elasticsearch/ingest_pipeline/default.yml:106]: i don't think we can make that change here, please can we remove from this PR. i think it needs more discussion.

File: packages/aws/kibana/dashboard/aws-fac28650-7349-11e9-816b-07687310a99a.json (hunk 6 above)
Comment (tommyers-elastic) [reply packages/aws/kibana/dashboard/aws-fac28650-7349-11e9-816b-07687310a99a.json:50]: in that case we could also add a query on datastream.type == metrics ?

File: packages/aws/data_stream/s3access/elasticsearch/ingest_pipeline/default.yml (hunk 4 above)
Comment (kaiyan-sheng) [reply packages/aws/data_stream/s3access/elasticsearch/ingest_pipeline/default.yml:106]: 😱 Can we add definition of these `geo` fields into `fields.yml` so we won't have mapping error anymore? Then we can discuss about this breaking change.

Comment (marc-gr) [reply packages/aws/data_stream/s3access/elasticsearch/ingest_pipeline/default.yml:106]: I added the definitions outside of the ECS fields, but also kept the new `client.geo.*` to start being compliant with ECS. We can decide how to procede about removing the root ones later on.

**Review by tommyers-elastic — APPROVED** [review]
LGTM - i think we could still simplify that dashboard filter to make it more maintainable, but not a blocker. thanks!
